client/src/app/transactions/transactions.component.ts

```diff
@@ -1,25 +1,48 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatSort} from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material';
+import { MatPaginator } from '@angular/material/paginator';
 
 import { TransactionsService } from './transactions.service';
 import { Transaction } from '../../../../src/transactions/transaction.entity';
 
+
 @Component({
   selector: 'app-transactions',
   templateUrl: './transactions.component.html',
   styleUrls: ['./transactions.component.scss']
 })
 export class TransactionsComponent implements OnInit {
-  public dbColumns: string[] = ['name', 'description', 'amount', 'date'];
+  public dbColumns: string[] = ['name', 'description', 'category', 'amount', 'date'];
   public displayedColumns: string[] = [...this.dbColumns, 'actions'];
-  public transactions$: Observable<Transaction[]> = this.transactionsService.getAll();
+  public transactions: Transaction[]; 
+  public dataSource: MatTableDataSource<Transaction>;
+
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
 
   constructor(private transactionsService: TransactionsService) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.dataSource = new MatTableDataSource();
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.transactionsService.getAll()
+      .subscribe(transactions => {
+        this.dataSource.data = this.transactions = transactions;
+      });
+  }
 
   capitalizeFirstLetter(str: string) {
     return str.charAt(0).toUpperCase() + str.slice(1);
   }
 
-}
+  delete(element) {
```

Comment:
This is already added as part of another PR and is not related to this issue. I would suggest rebasing on the current master where I have merged the other PR.